### PR TITLE
Embedded

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: doc/user/interop.md [hunk 1]
```diff
@@ -486,12 +487,30 @@ special-form, not a method.
...
-threads with the option `-Xsingle_threaded`, or
-`-Dtruffleruby.single_threaded=true` from another launcher.
+Ruby is designed to be a multi-threaded language and much of the ecosystem
+expects threads to be available. This may be incompatible with other Truffle
+languages which do not support threading, so you can disable the creation of
+multiple threads with the option `-Xsingle_threaded`. This option is set by
+default unless the Ruby launcher is used.
+
+It's separate from normal Ruby options and the
+[embedded configuration](#embedded-configuration) due technical details in the
```
Comment (nirvdrum) [inline doc/user/interop.md]: "due technical" -> "due to technical"

File: doc/user/interop.md [hunk 2]
```diff
@@ -486,12 +487,30 @@ special-form, not a method.
...
 but will not run the handler, as both of these features would require starting
 new threads.
+
+## Embedded configuration
+
+When used outside of the Ruby launcher, such as from another language's launcher
+via interop, embedded using the native polyglot library, or embedded via the
+Graal SDK, TruffleRuby will be automatically configured to work more
+cooperatively within another application. This includes options such as not
+installing an interrupt signal handler, and using the IO streams from the Graal
```
Comment (nirvdrum) [inline doc/user/interop.md]: I don't think the comma is necessary.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,10 @@
 
 New features:
 
-* ...
+* A new embedded configuration is used when TruffleRuby is used from another
+language or application. This disables features like signals which may conflict
+with the embedding application, and enables features like the use of polyglot IO
```
Comment (nirvdrum) [inline CHANGELOG.md]: I don't think this comma is necessary.

**Comment by eregon** [comment]
This is exactly the same as #1146 it seems.

**Comment by chrisseaton** [comment]
Yes I tried to put a change log entry in and then merge. That didn't work and I got a lot of weird failures. So I went back to your approach. I still don't know why it was failing, but the failures still appear in #1149.

**Pull request merged**